src/components/input/input.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * Grow Elastic TextArea
+ *
+ * Increases the `rows` attribute of a `textarea` until it no longer scrolls.
+ *
+ * @param {HTMLTextAreaElement} textarea - the target `textarea` element
+ */
+const growElasticTextArea = (textarea: HTMLTextAreaElement) => {
+  const minRows = Number(textarea.dataset.minRows);
+  let rows = Number(textarea.getAttribute('rows')) || minRows;
+  let isScrolling = textarea.scrollHeight > textarea.clientHeight;
+
+  // Grow until we stop scrolling
+  while (isScrolling) {
+    rows += 1;
+    textarea.setAttribute('rows', String(rows));
+    isScrolling = textarea.scrollHeight > textarea.clientHeight;
+  }
+};
+
+/**
+ * Shrink Elastic TextArea
+ *
+ * Decreases the `rows` attribute of a `textarea` until it begins scrolling.
+ *
+ * @param {HTMLTextAreaElement} textarea - the target `textarea` element
+ */
+const shrinkElasticTextArea = (textarea: HTMLTextAreaElement) => {
```

Comment:
Is the doubling up of the JSDoc `@param {HTMLTextAreaElement}` and TS `textarea: HTMLTextAreaElement` necessary? It feels like a bit much, but what do others think? 🤔

Reply:
Yeah I think we can remove it https://github.com/cloudfour/cloudfour.com-patterns/pull/716#discussion_r428228726

Reply:
I agree it looks like a bit much when reading the code.

That said I don't know a lot about how TS and JSDoc work. I would want both the type checking, and the param hint when using the function. Can we remove one of these and keep both of those features?

Reply:
Yep! VSCode will include the type from the TS type annotation, and the description from the JSDoc